app/base.go

```diff
@@ -14,27 +14,29 @@ type BaseApp struct {
 	decoder weave.TxDecoder
 	handler weave.Handler
 	ticker  weave.Ticker
+	debug   bool
```

Comment:
Yeah, I guess this is the most appropriate place to store the debug state. Good call.